lib/tasks/cms/cms.rake

```diff
@@ -210,6 +210,17 @@ namespace :cms do
         puts "\t++ Fixing pagination href in #{page.id}/shared/pagination"
       end
     end
+
+    desc 'Fixes empty titles in sections'
+    task :section_empty_titles => :environment do
+      CMS::Section.unscoped.where(title: ['', nil]).in_batches do |sections|
+        sections.each do |section|
+          title = section.system_name.presence || "Section #{section.id}"
+          section.update_column(:title, title)
+        end
+        sleep(0.01) # throttle
```

Comment:
IMO it's a bad idea to hardcode SQL queries. Using rails we can write the same code for alls DB engines.

About the throttle, AIUI it's necessary to allow the DB to perform other operations during the backfilling, so I left it.

Reply:
I wast thinking: what if, by a unlucky coincidence, the new `title` doesn't meet the uniqueness constraint? That would raise an exception and abort the script. It would be better to always set a UUID for title, so it reduces the chances of `title` being in use.

Reply:
This likely is a part of a transaction. When introducing a delay, that will just keep the transaction locks for longer, thus reducing the parallel operations the database can handle as well occupy database resources for longer.

Even if for a particular RDBMS it is of any advantage, I would classify it as a premature optimization unless you observed positive effects on the preview database.

I think the `update_all` would be the least resource intensive approach for both - database and client.

See https://stackoverflow.com/a/73038493/520567 , we can have something like

```
CMS::Section.unscoped.where(title: ['', nil]).update_all("title = system_name")
CMS::Section.unscoped.where(title: ['', nil]).update_all("title = id")
```

Although given how few elements we update, it is fine to leave current implementation and just remove the artificial delay.

Reply:
Is it possible to create a section without giving it a title? Is this task useful _right now_?

Reply:
Since I'm not opening a transaction explicitly, I assumed that would run out of a transaction. Is a transaction created automatically? Anyway I committed some changes.

Reply:
Yes, afaik everything runs by default in a transaction in porta.